R/utils.R

```diff
@@ -62,3 +61,24 @@ is_file_url <- function(x, error_call = caller_env()) {
 is_file_local <- function(x) {
   all(file.exists(x) & !dir.exists(x))
 }
+
+#' Convert an object or vector to a list
+#'
+#' If the input is already a list, it returns the input unchanged. If the input
+#' is a vector, it converts the vector to a list where each element of the vector
+#' becomes an element of the list. For all other types, it wraps the input in a
+#' list.
+#'
+#' @param x Any R object or vector.
+#' @return A list containing `x`, a list of the elements of `x` if `x` is a
+#' vector, or `x` itself if it's already a list.
+#'
+#' @keywords internal
+to_list <- function(x) {
```

Comment:
This is not needed, instead, use as.list() directly

```r
myvector <- c(1,2,3,4)

mylist <- list(1,2,3,4)

all.equal(as.list(myvector), as.list(mylist))

TRUE
```

Reply:
It is still necessary when working with R6 classes:

```
mylist <- list(rifampicin)  #creates a list
mylist2 <- as.list(rifampicin) #converts R6 itself into a list
all.equal(mylist, mylist2)

> Error in unclass(current) : cannot unclass an environment


mylist <- to_list(rifampicin)
mylist2 <- to_list(rifampicin)
all.equal(mylist, mylist2)

> TRUE
```

Reply:
This is true with only one element indeed.